vault/models/vault_entry.py

```diff
@@ -71,10 +75,25 @@ def _compute_complete_name(self):
 
     @api.model
     def search_panel_select_range(self, field_name):
-        """Add context to show just the name in searchpanel instead of full path"""
-        return super(
-            VaultEntry, self.with_context(entry_short_name=True)
-        ).search_panel_select_range(field_name)
+        """Add context to show just the name in searchpanel instead of full path.
+        We need to override function to prevent 200 records limit in search panel."""
+        old_value = models.SEARCH_PANEL_LIMIT
+        models.SEARCH_PANEL_LIMIT = False
+        _self = self.with_context(
+            search_panel_domain_override=True, entry_short_name=True
+        )
+        res = super(VaultEntry, _self).search_panel_select_range(field_name)
+        models.SEARCH_PANEL_LIMIT = old_value
+        return res
```

Comment:
I get the meaning behind it but the fact that you already did that here in multiple locations sounds like a required core patch instead of a hack like this in every location.

I would rather suggest to patch odoo.addons.web.models.models like this:
```
search_panel_limit = self.env.context.get("search_panel_limit", SEARCH_PANEL_LIMIT)
return {
    'parent_field': parent_name,
    'values': Comodel.with_context(hierarchical_naming=False).search_read([], fields, limit=search_panel_limit),
}
```
and set the context here.

Reply:
Finally, this is not necessary.

I have changed the approach in another way to simplify code.